src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -146,22 +152,52 @@ public ReadsDataSource( final List<File> samFiles, SamReaderFactory customSamRea
      *
      * @param intervals Our next full traversal will return only reads overlapping these intervals
      */
-    public void setIntervalsForTraversal( final List<SimpleInterval> intervals ){
-        // Treat null and empty interval lists the same
-        this.intervals = (intervals != null && ! intervals.isEmpty()) ? intervals : null;
+    public void setTraversalBounds( final List<SimpleInterval> intervals ) {
```

Comment:
For backwards compatibility reasons, since `ReadWalker` is currently the only unmapped-aware traversal. I'll see if I can refactor it away now, though.